config-templates/ibm/ibm-object-storage-plugin/2.2/custom-parameters.json

```diff
@@ -33,13 +33,20 @@
         "displayname": "COS Endpoint",
         "name": "cos-endpoint",
         "category": "config",
-        "required": "true"
+        "required": "false"
     },
     {
-        "description": "Enter COS storage class. For more info, refer to https://ibm.biz/cos-storage-classes",
+        "description": "Enter COS storage class. With cos-storageclass, provide value for either cos-endpoint or s3Provider, precedence is given to storageClass and s3Provider. To To input endpoint explicitly, input only endpoint and storageClass and skip s3Provider. For more info, refer to https://ibm.biz/cos-storage-classes",
         "displayname": "COS storage class",
         "name": "cos-storageclass",
         "category": "config",
         "required": "true"
+    },
+    {
+        "description": "Enter COS s3Provider. Choose s3Provider as `aws`, `wasabi` or `ibm` object storage",
+        "displayname": "COS s3Provider",
+        "name": "cos-s3provider",
+        "category": "config",
+        "required": "false"
```

Comment:
I think we can add `supported-values` and `regex` fields with four supported values:
1. ""
2. "ibm"
3. "aws"
4. "wasabi"

Reply:
I will discuss with UI team how empty value in supported-values list would be populated in UI and share the details.

Reply:
Just had discussion with UI team.
So if there is a parameter which has list of supported values but it can also take empty value.
Then we have to mark that param as Optional along with `supported-values` field.

Reply:
Sure adding regex